From 8507232302cc1dd5390a371ff590fea62fe7a5a2 Mon Sep 17 00:00:00 2001
From: Freerk-Ole Zakfeld <fzakfeld@scaleuptech.com>
Date: Wed, 16 Sep 2026 11:07:01 +0200
Subject: [PATCH 3/3] Support proxy environment for grafana

Support a proxy environment for grafana, which can be configured
using `grafana_container_proxy`, which defaults to the value of
`container_proxy`

Change-Id: I3bb752a6ed26a24388451fab76a43b35cdc76803
Signed-off-by: Freerk-Ole Zakfeld <fzakfeld@scaleuptech.com>
---
 ansible/roles/grafana/defaults/main.yml | 2 ++
 ansible/roles/grafana/handlers/main.yml | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/ansible/roles/grafana/defaults/main.yml b/ansible/roles/grafana/defaults/main.yml
index a007505b2..e4d7d471d 100644
--- a/ansible/roles/grafana/defaults/main.yml
+++ b/ansible/roles/grafana/defaults/main.yml
@@ -8,6 +8,7 @@ grafana_services:
     volumes: "{{ grafana_default_volumes + grafana_extra_volumes }}"
     dimensions: "{{ grafana_dimensions }}"
     healthcheck: "{{ grafana_healthcheck }}"
+    environment: "{{ grafana_container_proxy }}"
     haproxy:
       grafana_server:
         enabled: "{{ enable_grafana }}"
@@ -115,6 +116,7 @@ grafana_default_volumes:
   - "{{ node_config_directory }}/grafana/:{{ container_config_directory }}/:ro"
   - "kolla_logs:/var/log/kolla/"
 grafana_extra_volumes: "{{ default_extra_volumes }}"
+grafana_container_proxy: "{{ container_proxy }}"
 
 grafana_start_first_node_delay: 10
 grafana_start_first_node_retries: 12
diff --git a/ansible/roles/grafana/handlers/main.yml b/ansible/roles/grafana/handlers/main.yml
index 7e66df2da..917220fab 100644
--- a/ansible/roles/grafana/handlers/main.yml
+++ b/ansible/roles/grafana/handlers/main.yml
@@ -12,6 +12,7 @@
     image: "{{ service.image }}"
     volumes: "{{ service.volumes }}"
     dimensions: "{{ service.dimensions }}"
+    environment: "{{ service.environment }}"
     healthcheck: "{{ service.healthcheck | default(omit) }}"
   when:
     - inventory_hostname == groups[service.group] | first
@@ -48,6 +49,7 @@
     image: "{{ service.image }}"
     volumes: "{{ service.volumes }}"
     dimensions: "{{ service.dimensions }}"
+    environment: "{{ service.environment }}"
     healthcheck: "{{ service.healthcheck | default(omit) }}"
   when:
     - inventory_hostname != groups[service.group] | first
-- 
2.39.3 (Apple Git-145)

